### PR TITLE
Changes on Fastrope

### PR DESCRIPTION
**Pull request by pokertour**
**When merged this pull request will:**
- I've change the fastroping, by implement 5 items (5 ropes lenght). 5 interaction for each rope lenght.
- The change is you have to take with you or in the chopper the rope to be able to Deploy Fastrope.

- Change : 
-- Create a CfgWeapons.hpp with the 5 differents rope.
-- Change the CfgVehicles.hpp with the new interaction for each rope lenght.
-- Change the config.cpp for the CfgWeapons
-- Change the Stringtables.xml for the new items and interaction text. Need somes translations.
-- Change the fnc_deployropes.sqf to change the lenght of rope when deploy and remove item when rope has been deployed.

Some pictures : http://imgur.com/a/LLqo3

i test it on Singleplayer, need some test in multiplayer

I'm on Slack if you want speak

Do not consider changes on other addons "Cargo", "Spectator" and "Zeus"

Sorry for my bad english :(

### CODE REVIEW

File: addons/fastroping/functions/fnc_deployRopes.sqf [hunk 1]
```diff
@@ -25,6 +25,75 @@ private _hookAttachment = _vehicle getVariable [QGVAR(FRIES), _vehicle];
 {
     private _ropeOrigin = _x;
     private _hook = QGVAR(helper) createVehicle [0, 0, 0];
+    switch (_ropelength) do
+    {
```
Comment (jonpas) [inline addons/fastroping/functions/fnc_deployRopes.sqf]: Opening braces on same line as per [Coding Guidelines](https://ace3mod.com/wiki/development/coding-guidelines.html).

File: addons/fastroping/functions/fnc_canDeployRopes.sqf [hunk 2]
```diff
@@ -29,4 +29,4 @@ _deploymentStage = _vehicle getVariable [QGVAR(deploymentStage), 0];
 {
     (_deploymentStage == 0 && {getText (_config >> QGVAR(onPrepare)) == ""}) ||
     {_deploymentStage == 2 && {getText (_config >> QGVAR(onPrepare)) != ""}}
-}
+}
```
Comment (jonpas) [inline addons/fastroping/functions/fnc_canDeployRopes.sqf]: Missing empty last line.

File: addons/fastroping/CfgWeapons.hpp [hunk 3]
```diff
@@ -0,0 +1,60 @@
+class CfgWeapons {
+    class InventoryItem_Base_F;
+	class ACE_ItemCore;
```
Comment (BaerMitUmlaut) [inline addons/fastroping/CfgWeapons.hpp]: Tab, a couple more in this file too.

Comment (pokertour) [inline addons/fastroping/CfgWeapons.hpp]: Tab = 4 spaces in this file i don't understand what you want :/

Comment (BaerMitUmlaut) [inline addons/fastroping/CfgWeapons.hpp]: Nope, there's definitely a tab here. Note​ how the indentation is weird in this file on Github.

Comment (SilentSpike) [inline addons/fastroping/CfgWeapons.hpp]: You need to convert existing tab characters, your editor probably has a command to do this

**Comment by Drofseh** [comment]
Will every single person need their own rope?

If so that seems a little silly to me. IRL you'd usually just have one rope at each exit point on the helo.

**Comment by pokertour** [comment]
@drofseh no you have two solutions : 
1st you need 1 Rope in the chopper inventory and everyone can deploy it
2nd one guy need 1 rope in his backpack and only him can deploy the rope for everybody

**Comment by Drofseh** [comment]
@pokertour 
Cool, I'm on board with it then.

File: addons/fastroping/CfgVehicles.hpp [hunk 4]
```diff
@@ -45,10 +45,38 @@ class CfgVehicles {
...
                 priority = 1;
             };
-            class ACE_deployRopes {
-                displayName = CSTRING(Interaction_deployRopes);
-                condition = QUOTE([ARR_2(_player, vehicle _player)] call FUNC(canDeployRopes));
-                statement = QUOTE([ARR_2(QUOTE(QGVAR(deployRopes)), [vehicle _player])] call CBA_fnc_serverEvent);
+            class ACE_deployRopes12 {
+                displayName = CSTRING(Interaction_deployRopes12);
+                condition = (('ACE_rope12' in (backpackItems _player)) || ('ACE_rope12' in (ItemCargo vehicle _player))) && [_player, vehicle _player] call FUNC(canDeployRopes);
+                statement = [QGVAR(deployRopes), [vehicle _player, 12]] call CBA_fnc_serverEvent;
```
Comment (dedmen) [inline addons/fastroping/CfgVehicles.hpp]: You are not using `QUOTE`

Comment (dedmen) [inline addons/fastroping/CfgVehicles.hpp]: Could also easilly macro these Actions to make this a little more compact

**Comment by jonpas** [comment]
Moving to 3.12 as requested changes have not been addressed yet.

**Comment by synixebrett** [comment]
What is the status on this?

**Comment by BaerMitUmlaut** [comment]
Take a look at the status tag maybe?

**Comment by synixebrett** [comment]
I saw those, I was more asking about the fact that nothing has happened in 8 months.

**Comment by dedmen** [comment]
@pokertour showed no activity on github since mid-2017

**Comment by pokertour** [comment]
@dedmen hi man, can you tell me what is wrong in my request ?

**Comment by BaerMitUmlaut** [comment]
Conflicting files:
- addons/cargo/stringtable.xml
- addons/fastroping/stringtable.xml
- addons/spectator/stringtable.xml
- addons/zeus/stringtable.xml

**Comment by pokertour** [comment]
how can i remove this 3 files from this pull request : 
- addons/cargo/stringtable.xml
- addons/spectator/stringtable.xml
- addons/zeus/stringtable.xml

**Comment by synixebrett** [comment]
https://stackoverflow.com/questions/38743912/git-remove-a-file-from-a-pull-request

**Pull request closed without merging**